### PR TITLE
Adding 'rawRequest' config option to typescript-generic-sdk

### PR DESCRIPTION
**Pull request by franzwarning**
## Description

Adding a `rawRequest: boolean` config option to the typescript-generic-sdk. This allows the requestor to return `data` and `errors` on the top level.

Related # (issue)
https://github.com/dotansimha/graphql-code-generator/issues/6854

## Type of change

- [ x ] New feature (non-breaking change which adds functionality)

## Checklist:

- [ x ] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings
- [ x ] I have added tests that prove my fix is effective or that my feature works
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/plugins/typescript/generic-sdk/src/visitor.ts [hunk 1]
```diff
@@ -69,9 +72,12 @@ export class GenericSdkVisitor extends ClientSideBaseVisitor<RawGenericSdkPlugin
           o.node.variableDefinitions.length === 0 ||
           o.node.variableDefinitions.every(v => v.type.kind !== Kind.NON_NULL_TYPE || v.defaultValue);
         const returnType = usingObservable && o.operationType === 'Subscription' ? 'Observable' : 'Promise';
+        const resultData = this.config.rawRequest
+          ? `{ data: ${o.operationResultType}, errors: any }`
```
Comment (n1ru4l) [inline packages/plugins/typescript/generic-sdk/src/visitor.ts:76]:
```suggestion
          ? `{ data?: ${o.operationResultType}, errors?: Array<{ message: string; extensions?: unknown }>, extensions?: unknown }`
```

Shouldn't this be the following instead?

**Review by n1ru4l — COMMENTED** [review]
I only see typing changes here, can you add tests that verify that the functionality works as expected?

File: packages/plugins/typescript/generic-sdk/src/visitor.ts (hunk 1 above)
Comment (franzwarning) [reply packages/plugins/typescript/generic-sdk/src/visitor.ts]: Added tests and updated this line as well

**Comment by franzwarning** [comment]
@n1ru4l updated w/ tests

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**